### DISCUSSION TITLE
Needed understanding of replication strategy in emqx

### DISCUSSION THREAD

**Discussion by Shamanth-Urs** [emqx/emqx#5718] [Q&A]
**Please describe your problem in detail, if necessary, you can upload the log file through the attachment**:
Hi everyone, we are planning to use emqx brokers in our production. But looking at our scale we have to go with multiple emqx clusters.
As emqx mainly solves the connection and routing, I wanted to understand how does replication works within cluster. Considering the scenario where a node goes down, what will happen to the clients connected to that particular node? Will the messages sent to that particular node would be lost? Does client have to reconnect again to new node in cluster or is there a replica node which would take over in that scenario?

**Comment by qzhuyan** [comment]
| where a node goes down, what will happen to the clients connected to that particular node? 
| Does client have to reconnect again to new node in cluster or is there a replica node which would take over in that scenario?

The session between client and broker is replicated in the cluster, if one node is down, the client could reconnect which will end up on an alive node, that node will take over the session if clean_start = False in mqtt `connect` msg.

| Will the messages sent to that particular node would be lost?
This is secured at MQTT protocol level, Any message sent to down node will be lost, since no "PUBACK" "SUBACK" will be received, client should treat the message undelivered.
